### DISCUSSION TITLE
Constrainedby clause does not work if Enable Instance API is ON

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#10807] [Q&A] [answered]
Hi, 

I have detected (I think) a bug in classes that may be selected in a "replaceable class .... constrainedby" declaration when "Enable Instance API" is ON. 

Defined MolecularBindingType class as:
```
replaceable class MolecularBindingType = linearPBPK constrainedby dummyPBPK annotation(
     choicesAllMatching = true);
```
where the classes that may be selected are:
```
model linearPBPK "Linear Molecular Binding with PBPK scope"
    extends dummyPBPK;
    extends linear;
  end linearPBPK;
```
and
```
model NullPBPK "Null mechanism for non-network chemical activiy with PBPK scope"
    extends dummyPBPK;
    extends NullMech;
  end NullPBPK;
```
When the "Enable Instance API" is off then the proper classes appear in the dialog box:
![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/f0b38af0-6c16-45b9-8f67-e1ca239a61c6)

Whereas, when "Enable Instance API" is ON, then the dialog list changes as shown:
![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/5ff7b2fb-0855-4537-8adf-d52dfde68437)

Is this a bug? (OMedit v1.22.0-dev-155-g28817ae75c)

Thank you!

**Comment by casella** [comment]
Looks like it. Please open an issue and, if possible, provide indications to reproduce the issue.

**Reply by mpradovelasco** [reply, marked as answer]
Hi @casella,

I have reviewed the issue and it was a mistake of CybSim, that was not detected by OpenModelica when the Enable Instance API" is OFF. 
The code:
```
replaceable class MolecularBindingType = linearPBPK constrainedby dummyPBPK annotation(
choicesAllMatching = true);
```
should be:
```
replaceable class MolecularBindingType = linearPBPK constrainedby ChemicalActivity.dummyPBPK annotation(
     choicesAllMatching = true);
```
In other case, other dummyPBPK inherited classes in scope are also compliant with the clause constrainedby. Therefore, OpenModelica works better with Enable Instance API ON, with regard to this issue. 

Thanks!
